tests/common_test.cpp

```diff
@@ -651,4 +656,27 @@ void BaseFixture::TestAllFPGA(fpga_objtype otype, bool loadbitstream,
   }
 }
 
+typedef bool (*mock_enable_irq_t)(bool );
+bool MOCK_enable_irq(bool enable)
+{
+  dlerror(); // clear errors
+  mock_enable_irq_t real_mock_enable_irq =
+	  (mock_enable_irq_t) dlsym(RTLD_DEFAULT, "mock_enable_irq");
+  char *err = dlerror();
+
+  if (err) {
+    std::cerr << "dlsym(\"mock_enable_irq\") failed: " << err << std::endl;
+    std::cerr << "Be sure that libmock.so is loaded for mock tests." << std::endl;
+    throw std::logic_error("mock_enable_irq");
+  }
+
+  if (!real_mock_enable_irq) {
+    std::cerr << "dlsym(\"mock_enable_irq\") failed: (NULL fn pointer)" << std::endl;
+    std::cerr << "Be sure that libmock.so is loaded for mock tests." << std::endl;
+    throw std::logic_error("mock_enable_irq is NULL");
+  }
+
+  return real_mock_enable_irq(enable);
+}
+
```

Comment:
Perhaps a little comment above this snippet of code that explains what it is doing...
It took me a little while to figure it out ⌛️

Reply:
Sure